src/misc/ansi_code_markup.cpp

```diff
@@ -25,97 +25,213 @@
 #include "support.h"
 #include "ansi_code_markup.h"
 
-class ColorParser {
+/*!
+ * \brief Represents a markup tag.
+ *
+ * The constructor takes the string of a tag name and its optional value, and
+ * attempts to construct a valid tag that can be easily queried for its
+ * properties.
+ *
+ * The tag will be valid if, and only if, valid() returns true. Otherwise the
+ * tag could not be recognized.
+ */
+class Tag {
 public:
+	enum class TagGroup {
+		InvalidGroup,
+		Colors,
+		Erasers,
+		Styles,
+		Misc,
+	};
+
+	enum class TagName {
+		InvalidName,
+		Color,
+		BGColor,
+		EraseL,
+		EraseS,
+		It,
+		Bold,
+		Ul,
+		Strike,
+		Blink,
+		Dim,
+		Hidden,
+		Inverse,
+		Reset,
+	};
+
+	enum class ColorName {
+		InvalidColor,
+		Black,
+		Red,
+		Green,
+		Yellow,
+		Blue,
+		Magenta,
+		Cyan,
+		White,
+		Default,
+	};
+
+	enum class EraseExtents {
+		InvalidExtents,
```

Comment:
In general, I think `Invalid` is good enough. 
We're using enum classes, so all enums need to be prefixed by their type anyway:

These are all perfectly understandable:
 - EraseExtents::Invalid
 - Color::Invalid
 - Group::Invalid